src/main/groovy/io/cify/runner/utils/CapabilityParser.groovy

```diff
@@ -68,25 +101,76 @@ class CapabilityParser {
                 for (int b = 0; b < browserCapabilitiesList.size(); b++) {
                     LazyMap browser = browserCapabilitiesList.get(b) as LazyMap
                     Capabilities capability = new Capabilities()
-                    capability.setIos(ios)
-                    capability.setAndroid(android)
-                    capability.setBrowser(browser)
-                    capabilitiesSet.add(capability)
+                    capability.getIos().add(ios)
+                    capability.getAndroid().add(android)
+                    capability.getBrowser().add(browser)
+                    capabilities.add(capability)
                 }
             }
         }
-        capabilitiesSet
+        capabilities
+    }
+
+    /**
+     * Gets capabilities from capabilities file content and puts each capability into a separate {@link Capabilities} object
+     *
+     * @param capabilitiesFileContent - capabilities file content
+     * @return the list of {@link Capabilities}
+     */
+    private static List getCapabilitiesOneByOne(LazyMap capabilitiesFileContent) {
+        List iOSCapabilitiesList = getCapabilitiesForType(capabilitiesFileContent, IOS)
+        List browserCapabilitiesList = getCapabilitiesForType(capabilitiesFileContent, BROWSER)
+        List androidCapabilitiesList = getCapabilitiesForType(capabilitiesFileContent, ANDROID)
+
+        List capabilities = []
+
+        iOSCapabilitiesList.each {
+            Capabilities capability = new Capabilities()
+            capability.getIos().add(it as LazyMap)
+            capabilities.add(capability)
+        }
+
+        androidCapabilitiesList.each {
+            Capabilities capability = new Capabilities()
+            capability.getIos().add(it as LazyMap)
+            capabilities.add(capability)
+        }
+
+        browserCapabilitiesList.each {
+            Capabilities capability = new Capabilities()
+            capability.getIos().add(it as LazyMap)
+            capabilities.add(capability)
+        }
+
+        capabilities
+    }
+
+    /**
+     * Gets capabilities from capabilities file content and puts them all to one {@link Capabilities} object
+     *
+     * @param capabilitiesFileContent - capabilities file content
+     * @return the list of {@link Capabilities}
+     */
+    private static List getCapabilitiesAllInOne(LazyMap capabilitiesFileContent) {
+        List capabilities = []
```

Comment:
Maybe it worth to check if capabilitiesFileContent is empty and return an empty list instead of throwing an execption?

Reply:
True, fixed